src/reputation/tasks.py

```diff
@@ -82,6 +82,40 @@ def create_author_contribution(contribution_type, user_id, unified_doc_id, objec
     Contribution.objects.bulk_create(contributions)
 
 
+def check_transaction_success(transaction_hash):
+    # Check if connected successfully
+    if not w3.is_connected():
+        return "Failed to connect to Ethereum node."
+
+    try:
+        # Get transaction receipt
+        tx_receipt = w3.eth.get_transaction_receipt(transaction_hash)
+
+        # Check if transaction was successful
+        return tx_receipt.status == 1
+    except Exception as e:
+        log_error(e)
+        print(e)
+        return False
+
+
+@periodic_task(
+    run_every=crontab(minute="*/5"),
+    priority=4,
+    queue=QUEUE_PURCHASES,
+)
+def check_deposits():
+    deposits = Deposit.objects.filter(paid_status=None)
+    for deposit in deposits:
+        amt = deposit.amount
+        user = deposit.user
+        transaction_successful = check_transaction_success(deposit.transaction_hash)
+        if transaction_successful:
+            distribution = Dist("DEPOSIT", amt, give_rep=False)
+            distributor = Distributor(distribution, user, user, time.time(), user)
+            distributor.distribute()
```

Comment:
Should we add a deposit update here to mark it as paid?
```suggestion
            distributor.distribute()
            deposit.set_paid()
```

Reply:
Oh is the distribute not enough?

Reply:
The distributor just creates the distribution object and pays it out, but it doesn't update the deposit object.

Reply:
Ah you're right, done